### PR TITLE
Feat(core): Support flat + highlight button

### LINKED ISSUE #93: Support flat button + highlight
Currently we are banning the use of `style={Button.styles.flat}` and `highlight={true}` at the same time:

https://github.com/moaijs/moai/blob/d0f010471aa14fd1b81b6f8c9a9609e8f92ca322/core/src/button/button.tsx#L103-L106

As we can see this is only a "soft" ban, which means we throw a run-time error to prevent the users from using them, without any technical limitation. In fact, previously we avoid this only because we haven't sure what should it look like. Now we have.

<hr />

To start with this issue, first see that the "style" prop of Button is not a string, but an object. This allows Moai's themes can extend and modify the styles properly. However, it's pretty simple, a ButtonStyle is simply a group of classes to be applied to the button tag under different situations:

https://github.com/moaijs/moai/blob/d0f010471aa14fd1b81b6f8c9a9609e8f92ca322/core/src/button/button.tsx#L11-L16

And there are 2 pre-defined styles: "outset" and "flat":

https://github.com/moaijs/moai/blob/d0f010471aa14fd1b81b6f8c9a9609e8f92ca322/core/src/button/button.tsx#L157-L170

Which come from these 2 CSS Module files:

https://github.com/moaijs/moai/blob/d0f010471aa14fd1b81b6f8c9a9609e8f92ca322/core/src/button/button.tsx#L8-L9

**Notice that the "flat" style does not have a class for "highlight" field.** Meanwhile, the "outset" style does, which is why outset buttons have highlight style, but flat button does not.

Our job here is to define the "highlight" CSS for the "flat" style, which should be coded inside "flat.module.css", and referenced at Button.styles.flat.highlight (replace the empty string).

The CSS is simple, it should set the button text to our primary color, something like this:

```css
:global(.light) .highlight {
	color: var(--highlight-7);
}
:global(.dark) .highlight {
	color: var(--highlight-5);
}
```

<hr />

To test your work, just run the storybook locally (yarn storybook), and look for the Button story, then set "style" to "flat" and "highlight" to "true". If it shows a blue text, you should PR!

### PR DESCRIPTION
**Pull request by lqt93**
Fixes #93

### CODE REVIEW

File: core/src/button/flat.module.css [hunk 1]
```diff
@@ -5,7 +5,18 @@
 	border: solid 1px transparent;
 }
 
+/* HIGHLIGHT */
+
+:global(.light) .highlight {
+	color: var(--highlight-7);
```
Comment (thien-do) [inline core/src/button/flat.module.css:11]: @lqt93 Sorry this was my mistake, it should be "highlight-6", not 7! Please help me change it :D

File: core/src/button/flat.module.css [hunk 2]
```diff
@@ -5,7 +5,18 @@
...
+}
+
+:global(.dark) .highlight {
+	color: var(--highlight-5);
+}
+
 /* INTERACTION */
+
```
Comment (thien-do) [inline core/src/button/flat.module.css:19]: Thank you for the extra mile!

**Review by thien-do — APPROVED** [review]
Thank you @lqt93 ! This is already very good! Just one last step, I think we should put this on the widget gallery too. As you can see the current button gallery has 3 columns:

https://github.com/moaijs/moai/blob/39f39afda2cede8981b00d2ccd9c30393a0408a0/core/src/_gallery/button/style.tsx#L8-L16

Please help me have another one that features flat + highlight button, just between the 2nd and 3rd columns. Something of both the 10 and 14 lines

**Comment by lqt93** [comment]
All done, pls merge it @thien-do

**Review by thien-do — APPROVED** [review]
This is a very good PR @lqt93 ! Thank you!!!

**Pull request merged**